### PR TITLE
Provide option to force use of system scrollbar styling

### PR DESCRIPTION
**Pull request by vmstan**
Adds the ability to opt-out of the styled scrollbars that are present in Safari and Chrome on desktop since #31445

image

The current styling should have no impact on Firefox desktop or mobile as-is, they use a different styling that is system generated with defined colors.

When opting out of the styling, Chrome will render based on platform/OS system settings, Safari should render a throbber based on macOS settings.

### CODE REVIEW

**Comment by vmstan** [comment]
Reworded both the font and scrollbar options for brevity/consistency.

**Review by ClearlyClaire — DISMISSED** [review]
Looks good, but should this also apply to Firefox?

File: app/javascript/styles/mastodon-light/diff.scss
Comment (ClearlyClaire) [inline app/javascript/styles/mastodon-light/diff.scss:1]: Why the changes here?

File: config/locales/simple_form.en.yml [hunk 1]
```diff
@@ -222,7 +223,8 @@ en:
         setting_expand_spoilers: Always expand posts marked with content warnings
         setting_hide_network: Hide your social graph
         setting_reduce_motion: Reduce motion in animations
-        setting_system_font_ui: Use system's default font
+        setting_system_font_ui: Use operating system default font
```
Comment (ClearlyClaire) [inline config/locales/simple_form.en.yml:226]: Is the string change here intentional?

Comment (vmstan) [reply config/locales/simple_form.en.yml:226]: Only to make them consisntent between the two descriptions. I've changed it not change this one and to format the scrollbar option similarly.

File: app/javascript/styles/mastodon-light/diff.scss
Comment (vmstan) [reply app/javascript/styles/mastodon-light/diff.scss:1]: That was a failed conflict merge on my part. Fixed.

File: app/javascript/styles/mastodon-light/diff.scss [hunk 2]
```diff
@@ -525,17 +525,6 @@ a.sparkline {
   }
 }
 
-@supports not selector(::-webkit-scrollbar) {
-  html {
-    scrollbar-color: rgba($action-button-color, 0.25)
-      var(--background-border-color);
-  }
-}
```
Comment (ClearlyClaire) [inline app/javascript/styles/mastodon-light/diff.scss:533]: Is this removal intended?

**Comment by mjankowski** [comment]
Seems like adding this as a user option should be sort of a last resort to resolving some other way?

**Comment by vmstan** [comment]
I consider it similar reasoning to the system font option.

**Comment by mjankowski** [comment]
> I consider it similar reasoning to the system font option.

Could it literally be that option? "Use system stuff"? That's probably a bad idea ... just hesistant to add more combinations of options.

File: app/javascript/styles/mastodon-light/diff.scss (hunk 2 above)
Comment (ClearlyClaire) [reply app/javascript/styles/mastodon-light/diff.scss:533]: @vmstan asking again about that, since I did not get an answer.

**Comment by renchap** [comment]
We haven't got negative feedback about the new scrollbars since the 4.3 final release, is this still a complaint users have?

**Comment by ClearlyClaire** [comment]
> We haven't got negative feedback about the new scrollbars since the 4.3 final release, is this still a complaint users have?

Well, https://github.com/mastodon/mastodon/issues/33022

File: app/javascript/styles/mastodon-light/diff.scss (hunk 2 above)
Comment (vmstan) [reply app/javascript/styles/mastodon-light/diff.scss:533]: Sorry I did not see this earlier. I will address this today.

Comment (vmstan) [reply app/javascript/styles/mastodon-light/diff.scss]: Fixed.

**Review by ClearlyClaire — APPROVED** [review]
